examples/gno.land/r/docs/commondao/commondao_proposals.gno

```diff
@@ -0,0 +1,101 @@
+package commondao
+
+import (
+	"std"
+	"time"
+
+	"gno.land/p/nt/commondao"
+)
+
+// Proposal implements the commondao.Proposal interface
+// It contains anything useful to you in a proposal and defines how the proposal behaves
+type Proposal struct {
+	TitleProp       string
+	DescriptionProp string
+	CallerProp      std.Address
+}
+
+// Title returns the title of the proposal
+func (p *Proposal) Title() string {
+	return p.TitleProp
+}
+
+// Body returns the full description of the proposal
+func (p *Proposal) Body() string {
+	return p.DescriptionProp
+}
+
+// VotingPeriod defines the length of time the proposal is open for voting
+// Another possible implementation would be to have a constructor
+// So that each proposal can implement its own duration and tally
+func (c *Proposal) VotingPeriod() time.Duration {
+	return 24 * time.Hour // Voting period is 24 hours here
+}
+
+// Tally determines whether the proposal passes based on voting results
+// There are multiple ways to tally votes, pick the one most suited to your use case
+func (c *Proposal) Tally(r commondao.ReadonlyVotingRecord, m commondao.MemberSet) (bool, error) {
+	yesCount := r.VoteCount(commondao.ChoiceYes)
+
+	memberCount := DAO.Members().Size()
+	fivePercent := (memberCount*5 + 99) / 100 // fivePercent calculate what is 5% of the DAO's member
+
+	return yesCount >= fivePercent, nil // The proposal is considered as "passed" as soon as 5% of members voted yes
+}
```

Comment:
Other possible approach could be for example to use plurality, meaning that proposal passes if there is at least a YES vote, and the number of YES votes is greater than the number of NO votes, so initially a single vote could potentially make the proposal pass:

```suggestion
func (c *Proposal) Tally(r commondao.ReadonlyVotingRecord, m commondao.MemberSet) (bool, error) {
	choice, success := SelectChoiceByPlurality(record.Readonly())
	if !success {
		return false, nil
	}
	return choice == commondao.ChoiceYes, nil
}
```